include/api/CInferenceModelDefinition.h

```diff
@@ -113,8 +113,21 @@ class API_EXPORT CTrainedModel : public CSerializableToJson {
 
     enum ETargetType { E_Classification, E_Regression };
 
+    class CSizeInfo : public CSerializableToJson {
+    public:
+        explicit CSizeInfo(const CTrainedModel& trainedModel);
+        virtual ~CSizeInfo() override = default;
+        void addToDocument(rapidjson::Value& parentObject, TRapidJsonWriter& writer) const override;
+        //! \return Expected number of operation for the model evaluation.
+        virtual std::size_t numOperations() const = 0;
+
+    private:
+        const CTrainedModel& m_TrainedModel;
+    };
+    using TSizeInfoUPtr = std::unique_ptr<CSizeInfo>;
+
 public:
-    ~CTrainedModel() override = default;
+    virtual ~CTrainedModel() override = default;
```

Comment:
nit: I don't think this is needed since the implicit destructor is virtual if the base class destructor is virtual:
```suggestion
```

Reply:
This is good to know! :rocket:  Had to dig deep into the standard to verify this. :monocle_face:

Reply:
You missed this one :).